client/src/components/ElementEditor/Header.js

```diff
@@ -66,31 +88,43 @@ class Header extends Component {
    * @returns {ActionMenuComponent|null}
    */
   renderActionsMenu() {
-    const { id, expandable, ActionMenuComponent } = this.props;
+    const { id, expandable, ActionMenuComponent, isPublished } = this.props;
 
     // Don't show the menu when inline editing is not enabled
     if (!expandable) {
       return null;
     }
 
     const deleteTitle = i18n._t('ElementHeader.DELETE', 'Delete');
-    const deleteButtonClassNames = classNames('dropdown-item', 'element-editor__actions-delete');
+    const publishTitle = i18n._t('ElementHeader.PUBLISH', 'Publish');
+    const actionItemButtonClassName = 'dropdown-item';
 
     return (
       <ActionMenuComponent
         id={`element-editor-actions-${id}`}
-        className="element-editor-header__actions-dropdown"
+        className={'element-editor-header__actions-dropdown'}
         dropdownMenuProps={{ right: true }}
         toggleCallback={(event) => event.stopPropagation()}
       >
         <button
           onClick={this.handleDelete}
           title={deleteTitle}
           type="button"
-          className={deleteButtonClassNames}
+          className={classNames('element-editor__actions-delete', actionItemButtonClassName)}
         >
           {deleteTitle}
         </button>
+
+        {!isPublished &&
```

Comment:
How about we move this button into its own method - `renderPublishButton()` - it can return null if `isPublished` is true

Reply:
Good idea.